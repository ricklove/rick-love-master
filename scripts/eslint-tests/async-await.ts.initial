/* eslint-disable no-console */
/* eslint-disable @typescript-eslint/no-unused-vars */
const allowNoAwaitInAsync = async () => { return true; };

const basicPromise = () => new Promise<boolean>((resolve) => { resolve(true); });
const requireAwaitOnPromise = async () => { return basicPromise(); };

const voidPromise = () => new Promise<void>((resolve) => { resolve(); });
async function requireAwaitOnVoidPromise() {
    await basicPromise();
    voidPromise();
    await voidPromise();
}

const voidAsync = async () => await voidPromise();
async function requireAwaitOnVoidAsync() {
    await basicPromise();
    voidAsync();
    await voidAsync();
}

const numAsync = async () => await Promise.resolve(42);
async function requireAwaitOnNumAsync() {
    await basicPromise();
    numAsync();
    await numAsync();
}

async function requireAwaitTryCatch() {
    try {
        basicPromise();
    } catch{
        console.log(`error`);
    }
}
