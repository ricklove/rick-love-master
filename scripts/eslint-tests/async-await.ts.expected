/* eslint-disable no-console */
/* eslint-disable @typescript-eslint/no-unused-vars */
const requireAwaitExists = async () => { await Promise.resolve(); return true; };

const basicPromise = () => new Promise<boolean>((resolve) => { resolve(true); });
const requireAwaitOnPromise = async () => { return await basicPromise(); };

const voidPromise = () => new Promise<void>((resolve) => { resolve(); });
async function requireAwaitOnVoidPromise() {
    await basicPromise();
    await voidPromise();
    await voidPromise();
}

const voidAsync = async () => await voidPromise();
async function requireAwaitOnVoidAsync() {
    await basicPromise();
    await voidAsync();
    await voidAsync();
}

const numAsync = async () => await Promise.resolve(42);
async function requireAwaitOnNumAsync() {
    await basicPromise();
    await numAsync();
    await numAsync();
}

async function requireAwaitTryCatch() {
    try {
        await basicPromise();
    } catch{
        console.log(`error`);
    }
}
